packages/core/src/server/index.ts

```diff
@@ -40,7 +40,7 @@ export class DevServer {
     public logger: Logger,
     options?: UserServerConfig
   ) {
-    this.config = normalizeDevServerOptions(options);
+    this.config = normalizeDevServerOptions(options, "development");
```

Comment:
Not really sure why this is needed, the config should have been normalised once already.

Reply:
The compilation config and dev server config are normalized separately. We may merge them as single `normalizeConfig`

Reply:
Aha ok. I'm agnostic, we can keep it as is